types/hapi/definitions/util/lifecycle.d.ts

```diff
@@ -0,0 +1,51 @@
+import * as Boom from "boom";
+import * as stream from "stream";
+import {Request, ResponseToolkit} from "hapi";
+
+export namespace Lifecycle {
+
+    /**
+     * Lifecycle methods are the interface between the framework and the application. Many of the request lifecycle steps:
+     * extensions, authentication, handlers, pre-handler methods, and failAction function values are lifecyle methods
+     * provided by the developer and executed by the framework.
+     * Each lifecycle method is a function with the signature await function(request, h, [err]) where:
+     * * request - the request object.
+     * * h - the response toolkit the handler must call to set a response and return control back to the framework.
+     * * err - an error object availble only when the method is used as a failAction value.
+     */
+    export interface Method {
+        (request: Request, h: ResponseToolkit): ReturnValue;
+        (request: Request, h: ResponseToolkit, err: Error): ReturnValue;
+    }
+
+    /**
+     * Each lifecycle method must return a value or a promise that resolves into a value. If a lifecycle method returns
+     * without a value or resolves to an undefined value, an Internal Server Error (500) error response is sent.
+     * [See docs](https://github.com/hapijs/hapi/blob/master/API.md#lifecycle-methods)
+     */
+    export type ReturnValue = ReturnValueTypes | (Promise<ReturnValueTypes>);
+    export type ReturnValueTypes =
+        (null | undefined | string | number | boolean) |
+        (Buffer) |
+        (Error | Boom.BoomError) |
+        (stream.Stream) |
+        (Object | Object[]) |
+        // a toolkit signal:
+        // a toolkit method response:
+        any; // TODO need review
```

Comment:
Probably remove `any` in favour of stricter types?

Reply:
Commit 244e1ee. Replaced "any" with other types after reviewing the documents again.